### PR TITLE
Fix networking v1beta1 doc gen

### PR DESCRIPTION
**Pull request by howardjohn**
There was no generation previously - seems like they were copied at a
point in time and as such have fallen out of date. This adds the correct
generation logic then runs it

### CODE REVIEW

File: networking/v1beta1/workload_entry.pb.html [hunk 1]
```diff
@@ -0,0 +1,348 @@
+---
+title: Workload Entry
+description: Configuration affecting VMs onboarded into the mesh.
+location: https://istio.io/docs/reference/config/networking/workload-entry.html
```
[the remark this answers is not in the document]
Comment (frankbu) [reply networking/v1beta1/workload_entry.pb.html:4]: What do you mean by "doesn't exist"? It's in both [v1alpha3](https://github.com/istio/api/blob/master/networking/v1alpha3/workload_entry.proto) and [v1beta](https://github.com/istio/api/blob/master/networking/v1beta1/workload_entry.proto). Seems like both have the (same) doc, which includes tabs for both versions, so it seems that the html could be generated from either one.

[the remark this answers is not in the document]
Comment (frankbu) [reply networking/v1beta1/workload_entry.pb.html:4]: The location of the generated html doc is specified in the header: https://github.com/istio/api/blob/fcafad11e0aaa2453ac13d61396ecfb08045ef1b/networking/v1beta1/workload_entry.proto#L22. Generating from either one would replace the doc, so I'm assuming that the v1beta1 version is the one that we should be updating and generating from now. Never generate html from the v1alpha3 version. Does that sound right?

[the remark this answers is not in the document]
Comment (frankbu) [reply networking/v1beta1/workload_entry.pb.html:4]: Actually I guess generating both is fine (need the other things), but we should only ever pull the v1beta1 version to istio/istio.io. Still implies that doc changes (for either version) needs to be made in the v1beta1 proto.

**Comment by jasonwzm** [comment]
The current doc generation is in v1alpha3 folder. This is intentional as we don't differentiate the versions in our doc url (e.g. https://preliminary.istio.io/docs/reference/config/networking/workload-entry/) All the updated doc should be up-to-date in v1alpha3 folder, thus v1beta1 folder does not have doc gen.

**Comment by howardjohn** [comment]
@jasonwzm if they are using alpha docs, we should not have beta docs in the repo - its confusing to have generated code that isn't generated. However, i don't think its true, because https://preliminary.istio.io/docs/reference/config/networking/virtual-service/ is using the beta docs. Compare the two in istio/api @ master - the Table of contents is completely different between the two versions prior to this PR, and you can see its using beta docs?

**Comment by frankbu** [comment]
@jasonwzm This is confusing then. So all doc updates (typos, example correction, etc.) for either version of the API need to be made in the v1alpha3 version of the .proto file to get the update on the istio.io site? Do we need to make the same change in the corresponding v1alpha3 file as well (i.e., keep two exact copies of the docs)? If that's the process for now, what's the longer term plan?

**Comment by jasonwzm** [comment]
@howardjohn there is no html doc files for v1beta1, what we have is only a tab selection of v1alpha3 and v1beta1 yaml snippets in the v1alpha3 doc files. I am looking into what happened there with virtual service page where it does not have the tabs, but other networking reference pages are looking fine. The generated v1alpha3 html doc in master is looking fine.

**Comment by jasonwzm** [comment]
@frankbu This is a short-term solution to v1beta1, as v1beta1 is basically the same as v1alpha3 in terms of fields and docs. The only difference is the version name in CRs. So we adopted the tab strategy to include v1beta1 yaml snippets in v1alpha3 docs. In the long term, if we have different fields for a new API, we might have different generated docs for new versions.

**Comment by howardjohn** [comment]
@jasonwzm that may be the intent but that is not reality as far as I can tell. See `./networking/v1beta1/virtual_service.pb.html`, which exists on master today, is used in istio.io docs, and is not up to date

**Comment by jasonwzm** [comment]
@howardjohn we should delete that. Do you want do it in this PR or I can use another PR to delete that?

**Comment by howardjohn** [comment]
Feel free to send another PR. We should make sure that it does not break the istio.io update job, because right now its pulling from beta - not sure if it will automagically go to alpha

**Comment by ericvn** [comment]
istio.io should only be pulling `*.pb.html` files from the repo and placing them in the `location:` specified. If you have multiple files with the same `location:` it's probably indeterminate which one might be picked.

**Comment by ericvn** [comment]
Doing a quick search I see some duplicates:
```
./istio.mesh.v1alpha1.pb.html:location:                                 https://istio.io/docs/reference/config/istio.mesh.v1alpha1.html
./mesh/v1alpha1/istio.mesh.v1alpha1.pb.html:location:                   https://istio.io/docs/reference/config/istio.mesh.v1alpha1.html

./networking/v1alpha3/virtual_service.pb.html:location:                 https://istio.io/docs/reference/config/networking/virtual-service.html
./networking/v1beta1/virtual_service.pb.html:location:                  https://istio.io/docs/reference/config/networking/virtual-service.html
./networking/v1alpha3/workload_entry.pb.html:location:                  
```

**Comment by ericvn** [comment]
So remove the location from the ./networking/.../virtual_service.proto so it's removed from the pb.html you don't want copied, and then decide if the root is the correct istio.mesh.v1alpha1.pb.html to remove (the one is the directory has a .proto so expecting that one to be correct?).

**Comment by jasonwzm** [comment]
Removed the redundant docs file in https://github.com/istio/api/pull/1409. Closing this one.

**Pull request closed without merging**